Doc/c-api/type.rst

```diff
@@ -260,4 +260,7 @@ The following functions and structs are used to create
       The desired value of the slot. In most cases, this is a pointer
       to a function.
 
-      May not be ``NULL``.
+     .. versionchanged:: 3.10
+
+        The ``Py_tp_doc`` slot now can accepts ``NULL`` value.
+        Previously, the slot may not be ``NULL``.
```

Comment:
I would prefer to add the note to https://docs.python.org/dev/c-api/type.html?highlight=pytype_frommoduleandspec#c.PyType_FromModuleAndSpec documentation.